Tests/MongoSwiftSyncTests/UnifiedTestRunner/Matching.swift

```diff
@@ -0,0 +1,311 @@
+import MongoSwiftSync
+import TestsCommon
+
+/// Enum representing types that can be matched against expected values.
+enum MatchableResult {
+    /// A root document. i.e. a documents where extra keys are ignored when matching against an expected document.
+    case rootDocument(BSONDocument)
+    /// An array of root documents.
+    case rootDocumentArray([BSONDocument])
+    /// A (non-root) document.
+    case document(BSONDocument)
```

Comment:
what about naming this `subdocument` or `subDocument` to indicate it is non-root?